cmd/backup-manager/app/compact/manager.go

```diff
@@ -0,0 +1,229 @@
+}
+
+// logLine is line of JSON log.
+// It just extracted the message from the JSON and keeps the origin json bytes.
+// So you may extract fields from it by `json.Unmarshal(l.Raw, ...)`.
+type logLine struct {
+	Message string          `json:"Message"`
+	Raw     json.RawMessage `json:"-"`
+}
+
+// Manager mainly used to manage backup related work
+type Manager struct {
+	compact        *v1alpha1.CompactBackup
+	resourceLister listers.CompactBackupLister
+	statusUpdater  *CompactStatusUpdater
+	options        options.CompactOpts
+}
+
+// NewManager return a Manager
+func NewManager(
+	lister listers.CompactBackupLister,
+	statusUpdater *CompactStatusUpdater,
+	compactOpts options.CompactOpts) *Manager {
+	compact, err := lister.CompactBackups(compactOpts.Namespace).Get(compactOpts.ResourceName)
+	if err != nil {
+		klog.Errorf("can't find compact %s:%s CRD object, err: %v", compactOpts.Namespace, compactOpts.ResourceName, err)
+		return nil
+	}
+	return &Manager{
+		compact,
+		lister,
+		statusUpdater,
+		compactOpts,
+	}
+}
+
+func (cm *Manager) brBin() string {
+	return filepath.Join(util.BRBinPath, "br")
+}
+
+func (cm *Manager) kvCtlBin() string {
+	return filepath.Join(util.KVCTLBinPath, "tikv-ctl")
+}
+
+// ProcessBackup used to process the backup logic
+func (cm *Manager) ProcessCompact() error {
+	ctx, cancel := backuputil.GetContextForTerminationSignals(cm.options.ResourceName)
+	defer cancel()
+
+	compact, err := cm.resourceLister.CompactBackups(cm.options.Namespace).Get(cm.options.ResourceName)
+	defer func() { cm.statusUpdater.OnFinish(ctx, err, cm.compact) }()
+	if err != nil {
+		return errors.New("backup not found")
+	}
+	if err := options.ParseCompactOptions(compact, &cm.options); err != nil {
+		return errors.Annotate(err, "failed to parse compact options")
+	}
+
+	b64, err := cm.base64ifyStorage(ctx)
+	if err != nil {
+		return errors.Annotate(err, "failed to base64ify storage")
+	}
+	return cm.runCompaction(ctx, b64)
+}
+
+func (cm *Manager) base64ifyStorage(ctx context.Context) (string, error) {
+	brCmd, err := cm.base64ifyCmd(ctx)
+	if err != nil {
+		return "", err
+	}
+	out, err := brCmd.Output()
+	if err != nil {
+		eerr := err.(*exec.ExitError)
+		klog.Warningf("Failed to execute base64ify; stderr = %s", string(eerr.Stderr))
+		return "", errors.Annotatef(err, "failed to execute BR with args %v", brCmd.Args)
+	}
+	out = bytes.Trim(out, "\r\n \t")
+	return string(out), nil
+}
+
+func (cm *Manager) base64ifyCmd(ctx context.Context) (*exec.Cmd, error) {
+	br := cm.brBin()
+	args := []string{
+		"operator",
+		"base64ify",
+	}
+	StorageOpts, err := pkgutil.GenStorageArgsForFlag(cm.compact.Spec.StorageProvider, "storage")
+	if err != nil {
+		return nil, err
+	}
+	args = append(args, StorageOpts...)
+	return exec.CommandContext(ctx, br, args...), nil
+}
+
+func (cm *Manager) runCompaction(ctx context.Context, base64Storage string) (err error) {
+	cmd := cm.compactCmd(ctx, base64Storage)
+
+	logs, err := cmd.StderrPipe()
```

Comment:
Why use stderr instead of stdout? Another problem is use a more readable name to replace `logs`, such as `compactLogPipe`.

Reply:
Cause the tikv use stderr to print log, here is to receive the log